### ISSUE TITLE
Button to rerun failed tests in results view

### ISSUE THREAD

**Issue by jelovirt** [xspec/oXygen-XML-editor-xspec-support#17]
Add a button to results view to rerun only failed tests.

**Comment by AlexJitianu** [comment]
Currently, the XSpec way of running a set of particular tests is to set the **focus** attribute on them. I would prefer not to touch the XSpec files for this and to be able to pass as a parameter to the XSLTs the names of the scenarios to be executed. Or perhaps someone has a better idea about how this could be done....

**Comment by jelovirt** [comment]
When the XSpec file is converted to a XSLT with a unique mode per test, right? If you modify the compiler XSLT so that it's easy to determine which scenario matches which mode (e.g. using a counter), then you could run the XSLT starting immediately from the scenario specific mode.

(If you had this, then you could actually run e.g. 4 XSLT processes in parallel as an optimization.)

**Comment by AlexJitianu** [comment]
Indeed, the compiler generates a named template with an unique name for each scenario. I've overridden the name generation and I now generate an unique name using the scenario label as a seed  (the chances for two scenarios to have the same name are pretty slim so I've decided to leave it like this for now). The result is that I can control which templates are invoked through a driver XSLT that imports the compiled XSLT. I've also released a new version that contains this feature.
